File: elixir/log-level/lib/log_level.ex
defmodule LogLevel do
  def to_label(level, legacy?) do

    case {level, legacy?} do
      {0, false} -> :trace
      {1, _} -> :debug
      {2, _} -> :info
      {3, _} -> :warning
      {4, _} -> :error
      {5, false} -> :fatal
      _ -> :unknown
    end

  end

  def alert_recipient(level, legacy?) do

    lbl = to_label(level, legacy?)
    cond do
      lbl in [:error, :fatal] -> :ops
      lbl == :unknown and legacy? -> :dev1
      lbl == :unknown -> :dev2
      true -> false
    end

  end
end

File: java/logs-logs-logs/src/main/java/LogLevel.java
public enum LogLevel {
    UNKNOWN(0),
    TRACE(1),
    DEBUG(2),
    INFO(4),
    WARNING(5),
    ERROR(6),
    FATAL(42);
    private final int value;

    LogLevel(int value) {
        this.value = value;
    }

    public static LogLevel from(String logLevelText) {
        return switch (logLevelText) {
            case "TRC" -> LogLevel.TRACE;
            case "DBG" -> LogLevel.DEBUG;
            case "INF" -> LogLevel.INFO;
            case "WRN" -> LogLevel.WARNING;
            case "ERR" -> LogLevel.ERROR;
            case "FTL" -> LogLevel.FATAL;
            default -> LogLevel.UNKNOWN;
        };
    }

    public int getValue() {
        return value;
    }
}
